dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactory.java

```diff
@@ -286,4 +288,38 @@ public abstract class CategoryFactory {
 	 * @return List of Category filteredx
 	 */
 	public abstract List<HierarchedCategory> findAll(final CategorySearchCriteria searchCriteria) throws DotDataException;
+
+	/**
+	 * Find Categories by inodes and calculate its Hierarchy.
+	 *
+	 * For Example if you have the follows categories:
+	 *
+	 *
+	 * | Inode  | Name        |  key        |Parent          |
+	 * |--------|-------------|-------------|----------------|
+	 * | 1      | Top Category| top         | null           |
+	 * | 2      | Child       | child       | Top Category   |
+	 * | 3      | Grand Child | grand_child | Child          |
+	 *
+	 * And you find by inode 3 then you got:

```

Comment:
and you search** by inode

Reply:
done https://github.com/dotCMS/core/pull/29142/commits/7673fdbe9ce3848b65d22e4623eb37162652b0fa#diff-88379e3d0d9b5a5c54cb2207518bb686c3a9243cf6419ff11377616a7cd88dd2R304